.github/workflows/main.yml

```diff
@@ -0,0 +1,205 @@
+
+      - name: Run cargo-audit
+        run: pnpm rust:audit
+
+  semver_rust:
+    if: false # re-enable once spl-pod v0.5.1 is released
+    name: Check semver Rust
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cli: true
+          cargo-cache-key: cargo-semver
+
+      - name: Install cargo-semver-checks
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-semver-checks
+
+      - name: Run semver checks
+        run: pnpm rust:semver
+
+  spellcheck_rust:
+    name: Spellcheck Rust
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cargo-cache-key: cargo-spellcheck
+
+      - name: Install cargo-spellcheck
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-spellcheck
+
+      - name: Run cargo-spellcheck
+        run: pnpm rust:spellcheck
+
+  build_programs:
+    name: Build programs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cargo-cache-key: cargo-programs
+          solana: true
+
+      - name: Build Program
+        run: pnpm programs:build
+
+      - name: Upload Program Builds
+        uses: actions/upload-artifact@v4
+        with:
+          name: program-builds
+          path: ./target/deploy/*.so
+          if-no-files-found: error
+
+      - name: Save Program Builds For Client Jobs
+        uses: actions/cache/save@v4
+        with:
+          path: ./target/deploy/*.so
+          key: ${{ runner.os }}-builds-${{ github.sha }}
+
+  test_interface:
+    name: Test Interface
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cargo-cache-key: cargo-interface-tests
+          cargo-cache-fallback-key: cargo-interface
+
+      - name: Test
+        run: pnpm interface:test
+
+  test_programs:
+    name: Test Programs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cli: true
+          cargo-cache-key: cargo-program-tests
+          cargo-cache-fallback-key: cargo-programs
+          solana: true
+
+      - name: Test Programs
+        run: pnpm programs:test
+
+  test_client_js_legacy:
+    name: Test Client JS Legacy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+
```

Comment:
Are we not missing the restore program builds step here?

Reply:
The js tests don't actually need the built program, they're just doing codec checks